modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/GlobalSnapshotStorage.scala

```diff
@@ -65,3 +78,85 @@ object GlobalSnapshotStorage {
   case class InvalidGlobalSnapshotChain(expected: GlobalSnapshotLink, actual: GlobalSnapshotLink) extends NoStackTrace
 
 }
+
+object GlobalSnapshotStorageV2 {
+
+  def make[F[_]: Async](
+    globalSnapshotLocalFileStorage: GlobalSnapshotLocalFileStorage[F],
+    genesis: GlobalSnapshot,
+    inMemoryCapacity: PosInt
+  ): F[GlobalSnapshotStorageV2[F]] = {
+    def mkCacheQueue = Queue.unbounded[F, GlobalSnapshot]
+    def mkHeadRef = Ref.of[F, GlobalSnapshot](genesis)
+    def mkCache = MapRef.ofSingleImmutableMap[F, SnapshotOrdinal, GlobalSnapshot](Map.empty)
+    def mkOffloadQueue = Queue.unbounded[F, SnapshotOrdinal]
```

Comment:
Maybe we should store `Signed` or even `Hashed` `GlobalSnapshot`, of course if someone asks for it we should return `Signed[GlobalSnapshot]` but we could internally store it as `Hashed` and then we can compare hashes during update (if lastSnapshotHash matches)